internal/pkg/service/objectstorage/cloud/aws/iam.go

```diff
@@ -251,9 +254,9 @@ func templateRolePolicy(bucket *v1alpha1.Bucket) string {
 	return strings.ReplaceAll(rolePolicy, "@BUCKET_NAME@", bucket.Spec.Name)
 }
 
-func templateTrustPolicy(accountId string, managementCluster managementcluster.ManagementCluster, bucket *v1alpha1.Bucket) string {
-	policy := strings.ReplaceAll(trustIdentityPolicy, "@CLOUD_DOMAIN@", managementCluster.BaseDomain)
-	policy = strings.ReplaceAll(policy, "@INSTALLATION@", managementCluster.Name)
+func templateTrustPolicy(accountId string, baseDomain string, clusterName string, bucket *v1alpha1.Bucket) string {
```

Comment:
maybe we make this a function of IAMAccessRoleServiceAdapter so we can access the properties without passing them as arguments ?